spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/SparkReadConf.java

```diff
@@ -355,4 +355,21 @@ public boolean reportColumnStats() {
         .defaultValue(SparkSQLProperties.REPORT_COLUMN_STATS_DEFAULT)
         .parse();
   }
+
+  public boolean deriveStatsFromManifestSessionConf() {
+    return confParser
+        .booleanConf()
+        .sessionConf(SparkSQLProperties.DERIVE_STATS_FROM_MANIFEST_ENABLED)
+        .defaultValue(SparkSQLProperties.DERIVE_STATS_FROM_MANIFEST_ENABLED_DEFAULT)
+        .parse();
+  }
+
+  public boolean deriveStatsFromManifestTableProperty() {
```

Comment:
Why do we need two functions here? You should be able to add all the options into the same conf parser while still maintaining this hierarchy